server/utils/queryHelpers/scopeGet.js

```diff
@@ -89,18 +100,19 @@ getScopeElements = async (scopeInputs) => {
 		activeTargetType = 'collection';
 	}
 
-	if (!activeCommunity) {
+	if (!activeCommunity && communityId) {
 		activeCommunity = await Community.findOne({
 			where: { id: communityId },
 		});
 	}
```

Comment:
Given the note above, and that we don't need `communityId` to find an `activePub`, perhaps we simply remove this section and rely on the `Community.findOne` below with a slightly more general query:

```
if (!activeCommunity) {
	activeCommunity = await Community.findOne({
		where: { id: communityId || activePub.communityId },
	});
}
```

We would also have to move that function out of the `if (activeTargetType === 'pub') {` block to handle the case where `activeTargetType === 'collection'`

Reply:
Going to run with something similar to this, making it possible to scope the query to a community by leaving this block near the top of the function:

```js
if (!activeCommunity && communityId) {
    activeCommunity = await Community.findOne({
        where: { id: communityId },
    });
}
```
and then adding this one below (which is a more general version of what was living in the `activeTargetType === 'pub'` block:

```
if (!activeCommunity && activeTarget) {
    activeCommunity = await Community.findOne({
        where: { id: activeTarget.communityId },
    });
}
```

So the new behavior is, in psuedocode:

1. Query for `activeCommunity` if there is a `communityId` passed
2. Query for `activePub` (optionally scoped to `activeCommunity`)
3. Query for `activeCollection` (optionally scoped to `activeCommunity`)
4. If there is no `activeCommunity`, try to infer it from `activeTarget.communityId`

This makes me most confident that the various `getScope()` callsites that pass in a community ID where one is not strictly needed will continue to work as expected, but we can gradually move away from that as a best practice.